### PR TITLE
Allow users to bring their own HttpClient

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationContext.cs [hunk 1]
```diff
@@ -100,21 +98,43 @@ public AuthenticationContext(string authority, TokenCache tokenCache)
...
-        /// <param name="tokenCache">Token cache used to lookup cached tokens on calls to AcquireToken</param>
+        /// <param name="tokenCache">Token cache used to lookup cached tokens on calls to AcquireToken. Use <see cref="TokenCache.DefaultShared"/> 
+        /// to use ADAL's implementation of the cache on Android, iOS and UWP. Use null to store tokens only in memory. </param>
         public AuthenticationContext(string authority, bool validateAuthority, TokenCache tokenCache)
             : this(null, authority, validateAuthority ? AuthorityValidationType.True : AuthorityValidationType.False,
                 tokenCache)
         {
         }
 
+        /// <summary>
```
Comment (bgavrilMS) [inline src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationContext.cs:109]: @jmprieur - to review the comments

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationContext.cs [hunk 2]
```diff
@@ -100,21 +98,43 @@ public AuthenticationContext(string authority, TokenCache tokenCache)
...
+        /// <param name="authority">Address of the authority to issue token.</param>
+        /// <param name="validateAuthority">Flag to turn address validation ON or OFF.</param>
+        /// <param name="tokenCache">Token cache used to lookup cached tokens on calls to AcquireToken. Use <see cref="TokenCache.DefaultShared"/> 
+        /// to use ADAL's implementation of the cache on Android, iOS and UWP. Use null to store tokens only in memory.</param>
+        /// <param name="httpClient">Custom Http Client that will be used for the lifetime of this object</param>
+        public AuthenticationContext(string authority, bool validateAuthority, TokenCache tokenCache, HttpClient httpClient)
+            : this(null, authority, validateAuthority ? AuthorityValidationType.True : AuthorityValidationType.False,
+                   tokenCache, httpClient)
+        {
+            if (httpClient == null)
```
Comment (bgavrilMS) [inline src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationContext.cs:124]: If someone is using this ctor, they better pass in an httpClient. keeps things tidy.

**Comment by bgavrilMS** [comment]
Note that I am going to get this checked in so we can begin testing. We can change comments / make minor changes afterwards.

**Review by MarkZuber — APPROVED** [review]

**Comment by bgavrilMS** [comment]
Closing this because I merged it manually.

**Pull request closed without merging**